pkg/controller/scheduler.go

```diff
@@ -372,6 +373,15 @@ func (c *Controller) advanceCanary(name string, namespace string, skipLivenessCh
 	if cd.Spec.CanaryAnalysis.Iterations > 0 {
 		// increment iterations
 		if cd.Spec.CanaryAnalysis.Iterations > cd.Status.Iterations {
+			// If in "mirror" mode, mirror requests during the entire B/G canary test
+			if provider != "kubernetes" &&
```

Comment:
I'm only turning mirroring on if `provider != kubernetes` because down below we only route traffic to the canary (turn mirroring off) if `provider != kubernetes`.

I think this is fine and doesn't matter because `provider == kubernetes` is probably never going to work since you can't clone requests AFAIK.  But heads up, please LMK if there's a better way to do this.

Reply:
This is great, no need to call the mesh router if Kubernetes L4 is in use 👍